src/Signer/Key/InMemory.php

```diff
@@ -67,13 +67,17 @@ public static function file(string $path, string $passphrase = ''): self
         return new self($contents, $passphrase);
     }
 
-    public function contents(): string
+    public function sign(Signer $signer, string $payload): string
     {
-        return $this->contents;
+        $signer->validateAndConsumeKey($this->contents, $this->passphrase);
```

Comment:
`InMemory` (which should just be renamed as `Key` but I wanted to minimize PR diff) **must** now become a standalone class without interface, so we can enforce this `$signer->validateAndConsumeKey` call

Reply:
That makes no sense from a design PoV... a key doesn't sign anything.
We also remove the extension point for implementing JWK for no good reason.